### PR TITLE
added option to clear the whole cache after an web hook entry request

### PR DESCRIPTION
**Pull request by bestit-knoop**
Added the option to reset the whole content cache after an entry change request from the webhook system was detected.
This allows us to reset the whole cache without to include all cache keys in the consumer config. And this works also with dynamic cache keys that cant be added to the config.
This pr started as an requirement for an symfony 3.1 project where cache invalidation by tag is not avaible.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,6 +68,9 @@ best_it_contentful:
 
         # Which cache ids should be resetted everytime?
         collection_consumer:  []
+        
+        # Should the whole cache pool cleared after an entry reset over the webhook is detected
+        complete_clear_webhook: false
```
Comment (b3nl) [inline README.md]: "complete_clear_**on**_webhook" would be better ...

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -36,6 +36,10 @@ protected function getCachingConfig(): ArrayNodeDefinition
                     ->isRequired()
                     ->cannotBeEmpty()
                 ->end()
+                    ->booleanNode('complete_clear_webhook')
+                    ->info('Should the whole contentful cache cleared every time on an entry reset request?')
```
Comment (b3nl) [inline src/DependencyInjection/Configuration.php]: "... be cleared ... "

File: src/Service/CacheResetService.php [hunk 3]
```diff
@@ -28,15 +28,23 @@ class CacheResetService
...
      */
-    public function __construct(CacheItemPoolInterface $cache, array $cacheResetIds)
+    public function __construct(CacheItemPoolInterface $cache, array $cacheResetIds, bool $webhookCompleteReset)
     {
-        $this
-            ->setCache($cache)
-            ->setCacheResetIds($cacheResetIds);
+        $this->setCache($cache)
+            ->setCacheResetIds($cacheResetIds)
+            ->setWebhookCompleteReset($webhookCompleteReset);
```
Comment (b3nl) [inline src/Service/CacheResetService.php]: No need to use setters anymore. The team does not want to use them!

File: src/Service/CacheResetService.php [hunk 4]
```diff
@@ -58,6 +66,16 @@ protected function getCacheResetIds(): array
     }
 
     /**
+     * Is WebhookCompleteReset
+     *
+     * @return bool
+     */
+    protected function isWebhookCompleteReset(): bool
```
Comment (b3nl) [inline src/Service/CacheResetService.php]: You could remove this as well.

File: src/Service/CacheResetService.php (hunk 3 above)
Comment (bestit-knoop) [inline src/Service/CacheResetService.php]: To be honest i'am a bit confused here. Other Services in the bundle are using the setters and and in my opinion the pratice should be consistent across the same bundle.

**Comment by bestit-knoop** [comment]
Fixed config names and modified the readme

**Pull request merged**